reference/constraints/Compound.rst

```diff
@@ -0,0 +1,110 @@
+Compound
+========
+
+To the contrary to the other constraints, this constraint cannot be used on its own.
+Instead, it allows you to create your own set of reusable constraints, representing
+rules to use consistently across your application, by extending the constraint.
+
+==========  ===================================================================
+Applies to  :ref:`class <validation-class-target>` or :ref:`property or method <validation-property-target>`
+Options     - `groups`_
+            - `payload`_
+Class       :class:`Symfony\\Component\\Validator\\Constraints\\Compound`
+Validator   :class:`Symfony\\Component\\Validator\\Constraints\\CompoundValidator`
+==========  ===================================================================
+
+Basic Usage
+-----------
+
+Suppose that you have different places where a user password must be validated,
+you can create your own named set or requirements to be reused consistently everywhere:
+
+.. configuration-block::
+
+    .. code-block:: php
+
+        // src/Validator/Constraints/MatchesPasswordRequirements.php
+        namespace App\Validator\Constraints;
+
+        use Symfony\Component\Validator\Compound;
+
+        /**
+         * @Annotation
+         */
+        class MatchesPasswordRequirements extends Compound
+        {
+            protected function getConstraints(array $options): array
+            {
+                return [
+                    new NotBlank(),
+                    new Type('string'),
+                    new Length(['min' => 12]),
+                    new NotCompromisedPassword(),
```

Comment:
All those constraints needs a use on that file, I suggest to use an alias as usual, then: 
```php
use Symfony\Component\Validator\Constraints as Assert;
//...
return [
    new Assert\NotBlank(),
    // ...
```